sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/With.scala

```diff
@@ -92,6 +95,26 @@ object With {
     val commonExprRefs = commonExprDefs.map(new CommonExpressionRef(_))
     With(replaced(commonExprRefs), commonExprDefs)
   }
+
+  private def containsUnsupportedRef(
+    expr: Expression,
+    commonExprIds: Set[CommonExpressionId]
+  ): Boolean = {
+    expr match {
+      case _ if !expr.containsPattern(COMMON_EXPR_REF) => false
+      case w: With => containsUnsupportedAggExpr(w)
```

Comment:
Shall we remove this case? I think it's possible that the inner With has references to the outer With's common expressions, and we should detect that.

Reply:
good point, and we don't need to call `containsUnsupportedAggExpr` again on the inner With because we can't have nested aggregate expressions?

Reply:
updated the PR